decred/decred/dcr/rpc.py

```diff
@@ -80,6 +81,233 @@ def call(self, method, *params):
             raise Exception("%s error: %r" % (method, res["error"]))
         return res["result"]
 
+    def addNode(self, addr, subCmd):
+        """
+        Attempts to add or remove a persistent peer.
+
+        Args:
+            addr (str) IP address and port of the peer to operate on
+            subCmd (str) 'add' to add a persistent peer, 'remove' to remove a
+                persistent peer, or 'onetry' to try a single connection to a peer
+        """
+        self.call("addnode", addr, subCmd)
+
+    def createRawSSRTx(self, ticket, relayFee=None):
+        """
+        Returns a new unsigned transaction revoking the ticket.
+
+        Args:
+            ticket (msgTx.MsgTx): The ticket to revoke.
+            fee (int): Optional. Default=None. The fee to apply to the
+                revocation in atoms/kb.
+
+        Returns:
+            MstTx.msgtx: The revocation.
+        """
+        if relayFee:
+            outs = []
+            # Parse the ticket purchase transaction to determine the required output
+            # destinations for vote rewards or revocations.
+            (
+                ticketPayKinds,
+                ticketHash160s,
+                ticketValues,
+                _,
+                _,
+                _,
+            ) = txscript.sstxStakeOutputInfo(ticket.txOut)
+
+            # Calculate the output values for the revocation.  Revocations do not
+            # contain any subsidy.
+            values = txscript.calculateRewards(ticketValues, ticket.txOut[0].value, 0)
```

Comment:
I know these were added in the revocations PR, but I noticed that `sstxStakeOutputInfo` and `calculateRewards` don't have any tests. **dcrd** has `TestGetSStxStakeOutputInfo` and `TestGetStakeRewards`. I'll add an entry to #70.

Reply:
Thank you.